broker/src/main/java/io/camunda/zeebe/broker/logstreams/AtomixAppendListenerAdapter.java

```diff
@@ -5,18 +5,15 @@
  * Licensed under the Zeebe Community License 1.1. You may not use this file
  * except in compliance with the Zeebe Community License 1.1.
  */
-package io.camunda.zeebe.logstreams.storage.atomix;
+package io.camunda.zeebe.broker.logstreams;
 
 import io.atomix.raft.storage.log.IndexedRaftLogEntry;
 import io.atomix.raft.zeebe.ZeebeLogAppender.AppendListener;
 import io.camunda.zeebe.logstreams.storage.LogStorage;
 
-public final class AtomixAppendListenerAdapter implements AppendListener {
-  private final LogStorage.AppendListener delegate;
-
-  public AtomixAppendListenerAdapter(final LogStorage.AppendListener delegate) {
-    this.delegate = delegate;
-  }
+public record AtomixAppendListenerAdapter(
```

Comment:
:question: So I know we haven't talked about this in the team before. IntelliJ just tells you that something can be a record if all its properties are immutable, which is technically true. However, it's a little strange for me semantically. A record is a data/value class. See, for example, the Oracle docs:

> Record classes, which are a special kind of class, help to model plain data aggregates with less ceremony than normal classes.
>
>For background information about record classes, see JEP 395.
>
>A record declaration specifies in a header a description of its contents; the appropriate accessors, constructor, equals, hashCode, and toString methods are created automatically. A record's fields are final because the class is intended to serve as a simple "data carrier". 

I don't think the adapter is a data carrier here, so I would keep it as a normal class. I see the benefit in indicating intent, mostly.

This is fairly new though, so my thoughts on it are also fairly new :smile: Happy to discuss that

Reply:
Tbh I don't care. I just followed the instructions from intellij and thought we want to migrate most of our classes :shrug: I can revert this if you want.

Reply:
I would like to keep records for what they're meant to be, i.e. data carrier, since by default all their properties become public (which we may not always want). So let's do that. If you don't care, then let's keep doing that :wink: And when you do we can discuss it :smile: